### PR TITLE
Add types to shipit

### PR DESCRIPTION
**Pull request by pbiggar**
I started adding a very small feature to shipit and did not feel very confident doing it, so I added types to it. This makes no change to functionality, but pretty aggressively types everything.

My plan is to merge this then do a refactor to remove dups and use real types instead of dicts everywhere.

### CODE REVIEW

File: services/bwdserver-deployment/shipit.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ k8s:
 
     # If you change the nginx override, it needs a rolling restart
     custom-post-apply:
-      - kubectl rollout restart --namespace default deployment bwdserver-deployment
```
Comment (StachuDotNet) [inline services/bwdserver-deployment/shipit.yaml:10]: how did this work before? Does `kubectl` just assume the `/`?

File: scripts/deployment/shipit [hunk 2]
```diff
@@ -381,44 +511,43 @@ def containers_pull(cliargs):
...
-        'image': f"{url}:{image_id}",
-        'image_latest': f"{url}:latest"
-    })()
+    return {'image': f"{url}:{image_id}", 'image_latest': f"{url}:latest"}
 
   for c in get_service_containers(cliargs):
     info = image_info(c)
-    run("docker tag id", ["docker", "tag", f"{c}:latest", info.image])
-    run("docker tag latest", ["docker", "tag", f"{c}:latest", info.image_latest])
+    run("docker tag id", ["docker", "tag", f"{c}:latest", info["image"]])
```
Comment (StachuDotNet) [inline scripts/deployment/shipit:527]: are the changes from `.` to `["..."]` required to have type checks take place?

**Comment by StachuDotNet** [comment]
An update to `scripts/deployment/README.md` may be worthwhile. (if anything, a quick "coding guide"-like note would be sufficient)

**Comment by StachuDotNet** [comment]
This is looking great. Happy to see MyPy in the wild. Barring CI failures and some minor questions, I'll be happy to approve as-is. (README change non-blocking, as it's likely obvious - and for all I know, MyPy will complain if something lacks type annotations, rendering a "coding guide" comment useless.)

File: services/bwdserver-deployment/shipit.yaml (hunk 1 above)
Comment (pbiggar) [reply services/bwdserver-deployment/shipit.yaml:10]: yeah

**Pull request merged**